### PR TITLE
enhance: Support auto index for scalar index

### PR DESCRIPTION
**Pull request by xiaocai2333**
issue: #29309

### CODE REVIEW

File: internal/proxy/task_index.go [hunk 1]
```diff
@@ -141,28 +141,27 @@ func (cit *createIndexTask) parseIndexParams() error {
...
-			}
 
-			if exist && !validateStringIndexType(specifyIndexType) {
-				return merr.WrapErrParameterInvalid(DefaultStringIndexType, specifyIndexType, "index type not match")
-			}
-		} else if typeutil.IsArithmetic(cit.fieldSchema.DataType) {
-			if !exist {
-				indexParamsMap[common.IndexTypeKey] = DefaultArithmeticIndexType
-			}
+	indexType := indexParamsMap[common.IndexTypeKey]
```
Comment (czs007) [inline internal/proxy/task_index.go:145]: should check  common.IndexTypeKey exist ?

Comment (xiaocai2333) [reply internal/proxy/task_index.go:145]: If it not exist, the value is "".

**Comment by czs007** [comment]
close for now , reopen in the future.

**Pull request closed without merging**